### DISCUSSION TITLE
리팩터링 2판 회고

### DISCUSSION THREAD

**Discussion by choisohyun** [frontend-book-study/refactoring-2#20] [리팩터링 2판]
```
## 💡 개인적인 회고
### 독서 과정 돌아보기
- 읽기 전 기대했던 점:
- 실제로 발견한 점:
- 새롭게 배운 것들:

### 모임을 통해 얻은 것
- 혼자 읽었다면 놓쳤을 부분:
- 다른 사람들의 관점에서 배운 점:
- 앞으로 적용하고 싶은 것들:


### 모임 회고
**자유롭게 피드백**
- 
- 

**다음에 읽고싶은 책 / 스터디**
- 
- 
```

**Comment by blan19** [comment]
## 💡 개인적인 회고
### 독서 과정 돌아보기
- 읽기 전 기대했던 점: 거미줄 같은 내 코드들에게서 지도가 되어 줄 구원자
- 실제로 발견한 점: 알음 알음 알았던 나의 리팩토링 방법을 체계화 시켜주었습니다.
- 새롭게 배운 것들: 리팩토링도 체계화된 패턴이며, 그 패턴들의 집합으로 점진적으로 리팩토링을 적용할 수 있다.

### 모임을 통해 얻은 것
- 혼자 읽었다면 놓쳤을 부분: 매번 느끼지만 사람마다 다양한 시각이 존재했었고, 리팩토링 2판을 읽으면서 실습한 리팩토링한 좋은 코드들을 보며 많이 배웠습니다.
- 앞으로 적용하고 싶은 것들: 실무에서 지나친 레거시 코드들의 개선!


### 모임 회고
**오늘 모임의 특별했던 점**
- 최근에 매우 바빠졌지만 참여가 가능하다면 참여하여 다른 분들이 리팩토링에 대한 좋은 예시와 관점들을 말해주는 것을 듣고자했습니다.
- 매번 똑같은 환경속에 있으면 편협해지기 쉬운데, 좀 더 넓게 보는 시야를 주어 스스로 소프트하게 채찍질을 해주어 좋은 것 같습니다.

**다음에 읽고싶은 책 / 스터디**
- 자바스크립트 + 리액트 디자인 패턴
- 리액트 훅을 활용한 마이크로 상태 관리
- 프런트엔드 개발을 위한 테스트 입문

**Comment by RAINBOWJYP** [comment]
## 💡 개인적인 회고
### 독서 과정 돌아보기
- 읽기 전 기대했던 점: 멋지게 리팩터링하고 있을 미래의 나를 상상함
- 실제로 발견한 점: 여전히 허우적대고 있는 나를 발견함. 
- 새롭게 배운 것들: 테스트를 기계처럼 습관화해야 좋은 습관인 것을 알았다. 클래스가 익숙하지 않았는데 실습하면서 상속과 슈퍼클래스에 대해 이해하게 되었다. 곧 클래스를 활용한 프로젝트를 진행할 것 같은데 도움이 될 것 같다!

### 모임을 통해 얻은 것
- 혼자 읽었다면 놓쳤을 부분: 완주하지 못했을 것... 이해안되는 부분을 스터디원들이 해소시켜준 것 (최고)
- 다른 사람들의 관점에서 배운 점: 실무에서 리팩터링을 어떻게 하는지 또 이 책을 읽고 어떤 고민과 깨달음을 얻었는지 볼 수 있어서 좋았다.
- 앞으로 적용하고 싶은 것들: 중복되는 코드 하나로 묶기, 안쓰는 코드 죽이기, 이해하기 쉬운 이름으로 메서드명, 변수명 바꾸기 등 내가 이해했던 작은 것들부터 리팩터링하여 점차 성장해나가고 싶다!!

### 모임 회고
**자유롭게 피드백**
- 넘넘 좋아요. 완벽한데여!!!!!!!!! 최고 이 모임 덕분에 갓생사는 것 같고 뿌듯해용

**다음에 읽고싶은 책 / 스터디**
- 기술책말고 개발자 이야기? 같은 경험을 얘기해주는 책은 어떨까요??

**Comment by LastCow9000** [comment]
## 💡 개인적인 회고
### 독서 과정 돌아보기
- 읽기 전 기대했던 점: 리팩터링에 대한 팁
- 실제로 발견한 점: 기법들과 수행 방법, 가독성에 대한 팁, 객체 지향적 관점
- 새롭게 배운 것들: 문장 슬라이드의 가치, 잘못된 코딩 습관

### 모임을 통해 얻은 것
- 혼자 읽었다면 놓쳤을 부분: 완독하기가 힘들었을 것이다....
- 다른 사람들의 관점에서 배운 점: 같은 문장에 대한 깊은 사고와 다양한 관점
- 앞으로 적용하고 싶은 것들: 이 책에 나와있는 모든 것들을 습관적으로 사용할 수 있도록 하고 싶다!


### 모임 회고
**자유롭게 피드백**
- 혼자 읽었다면 중간에 하차 했을 법한데 덕분에 이번에도 한 권을 완독하게 되었네요!

**다음에 읽고싶은 책 / 스터디**
- 이번 책에서 나온 객체 지향에 대해 더 생각해볼 수 있도록 '객체 지향의 사실과 오해'
- 클린 코드, 디자인 패턴
- 타입스크립트 프로그래밍, 러닝 타입스크립트, 이펙티브 타입스크립트 (난이도 순)

**Comment by choisohyun** [comment]
## 💡 개인적인 회고
### 독서 과정 돌아보기

- 읽기 전 기대했던 점: 책에 있는 리팩토링을 현실 프로덕에 적용할 수 있다.
- 실제로 발견한 점: 아직 적용해본 부분은 없다..

### 모임을 통해 얻은 것

- 혼자 읽었다면 놓쳤을 부분: 명령 패턴 등 책에 잠시 나오는 부분에 대해 리뷰로 작성해주시거나 해서 좀 더 자세히 살펴볼 수 있었습니다.
- 다른 사람들의 관점에서 배운 점: 클래스나 리팩토링 기법들에 대한 다양한 얘기와 의견을 나눠볼 수 있었습니다.
- 앞으로 적용하고 싶은 것들: 현실에 리팩토링 적용하기, 테스트 범위 중요부분에 넓혀보기

### 모임 회고

**자유롭게 피드백**

- 실습은 생각보다 더 어렵고 시간이 많이 걸렸다. 계속 하려면 좀 더 디밸롭이 필요할듯
- 게시판 형태로 참여하는게 생각보다 좋아서 계속 쓰게 됐다. 블로그랑 어떻게 합칠지 좀더 고민할 필요가 있겠다.

**다음에 읽고싶은 책 / 스터디**

- 생성형 AI + 프론트엔드... 이런 책이 안보인다

**Comment by hamo-o** [comment]
## 💡 개인적인 회고
### 독서 과정 돌아보기
- 읽기 전 기대했던 점: 리팩터링에 대한 구체적인 전략 습득
- 실제로 발견한 점: 리팩터링을 언제 해야 하는지 (리팩터링 전반에 관한 내용들) + 리팩터링에 대한 구체적인 전략 습득
- 새롭게 배운 것들: 리팩터링은 모든 것을 다 끝내고 하는 것이 아니라, 무언가를 시작하기 전에 하는 것......

### 모임을 통해 얻은 것
- 혼자 읽었다면 놓쳤을 부분: 직접 리팩터링을 적용해보지 않아서 조금 추상적으로 느껴졌는데, 질답과 실습을 통해서 더 깊이 이해할 수 있었습니다!!
- 다른 사람들의 관점에서 배운 점: 모르는 게 나왔을 때나 조금 애매한 리팩터링인 것 같을 때 다른 분들의 생각을 들으면서 방향을 잡아나간 것 같아 좋았습니다.
- 앞으로 적용하고 싶은 것들: Chap6)기본적인 리팩터링, Chap8) 기능 이동, Chap10) 조건부 로직 간소화  이 세 챕터가 코드에서 가장 많이 등장한다고 생각이 들어서 먼저 적용해보고 싶습니다.


### 모임 회고
**자유롭게 피드백**
- 실습의 경우 딱 맞는 문제 찾기가 어려웠는데 앞선 스터디들의 자료를 어디선가 찾는다면.. 수월해질 것 같습니다
- 실습, 질답 모두 너무 좋았습니다!

**다음에 읽고싶은 책 / 스터디**
- 자바스크립트 + 리액트 디자인 패턴 (디자인 패턴 책이면 다른 도서도 좋습니다!)
- 프론트엔드 성능 최적화 가이드

**Comment by hyungwookchoi** [comment]
## 💡 개인적인 회고
### 독서 과정 돌아보기
- 읽기 전 기대했던 점: 여러가지 핑계로 미처 수정하지 못했던 사내 코드들.. 을 원칙에 맞춰 우아하게 리팩토링 하기
- 실제로 발견한 점: 그 동안 원칙 없는 리팩토링을 해오지 않았나? 책의 원칙을 따라서 리팩토링 해보면 좋을 것 같다.
- 새롭게 배운 것들: 리팩토링은 개발의 한 과정, 미루지 말고 조금씩이라도 개선하자.

### 모임을 통해 얻은 것
- 혼자 읽었다면 놓쳤을 부분: 이번에도 역시 책을 끝까지 완주할 수 있었다는 점이 가장 컸습니다.
- 다른 사람들의 관점에서 배운 점: 같은 코드에서 시작해서 리팩토링 하더라도 다른 결과가 나왔고, 각자의 생각을 엿볼 수 있어 좋았습니다.
- 앞으로 적용하고 싶은 것들: 작게라도 테스트 코드를 작성해서 커버리지 높이기


### 모임 회고
**자유롭게 피드백**
- 실습을 완벽하게 다 수행하긴 어려웠지만.. 책만 읽는 게 아니라 실제로 코드를 작성하면서 개선해보는 게 좋았습니다.

**다음에 읽고싶은 책 / 스터디**
- 윗분들이 작성해주신 JS, TS 혹은 리액트 관련 여러 책 다 좋아요.
- 소프트웨어 엔지니어 가이드북 (최근에 출간했는데 좋다는 평이 많고, 개발자 커리어에 대해 다루는 책이라 지금 딱 나한테 필요한 내용인 것 같아서...! 그런데 같이 읽어 볼 만한 내용인지는 잘 모르겠다.)

**Comment by suojae** [comment]
## 💡 개인적인 회고
### 독서 과정 돌아보기
- 읽기 전 기대했던 점: 리팩토링은 어디서부터 시작해야할까? 에 대한 궁금중 해소
- 실제로 발견한 점: 일단 테스트 코드부터 작성하자!
- 새롭게 배운 것들: 리팩토링 방법에 대한 현실적인 구체적인 스텝들

### 모임을 통해 얻은 것
- 혼자 읽었다면 놓쳤을 부분: 혼자 읽었다면 아 좋은 문구들이구나하고 뇌에 잠깐 스쳐지나갈 것들이었는데 질문과 답을 통해 뇌에 새길 수 있었다.
- 다른 사람들의 관점에서 배운 점: 책 내용에 약간 반대하고 싶은 내용도 있었는데 단순히 혼자 생각하는 것에서 멈추지 않고 토론을 할 수 있어서 견해를 넓힐 수 있었다.
- 앞으로 적용하고 싶은 것들: 서로 질의-딥이 너무 좋았어서.. 넵  추가적으로 적용하고 싶은 것들은 책마다 다를 것 같습니다


### 모임 회고
**자유롭게 피드백**
- 스터디장님이 항상 인도(?)해주셔서 너무감사했습니다🥹
- 책읽기도 프로젝트도 역시 함께하는게 최고, 함께자라기 너무 좋아요

**다음에 읽고싶은 책 / 스터디**
- 이펙티브 타입스크립트
